tests/baselines/reference/narrowingDestructuring.types

```diff
@@ -144,7 +144,7 @@ function farr<T extends [number, string, string] | [string, number, number]>(x:
 
         const [head, ...tail] = x;
 >head : "number"

```

Comment:
a little bit outside of the scope of this PR but most likely the `if` statement related to this line should use `typeof` and this type here should be `number` and not `"number"`